crates/services/bls_aggregation/src/bls_agg.rs

```diff
@@ -303,117 +342,177 @@ impl<A: AvsRegistryService + Send + Sync + Clone + 'static> BlsAggregatorService
+        tokio::pin!(task_expired_timer);
+
+        loop {
+            tokio::select! {
+                _ = &mut task_expired_timer => {
+                    // Task expired. If window is open, send aggregated reponse. Else, send error
+                    if open_window {
+                        aggregated_response_sender
+                            .send(Ok(current_aggregated_response.unwrap()))
+                            .map_err(|_| BlsAggregationServiceError::ChannelError)?;
+                    } else {
+                        let _ = aggregated_response_sender.send(Err(BlsAggregationServiceError::TaskExpired));
+                    }
+                    return Ok(());
+                },
+                _ = window_rx.recv() => {
+                    // Window finished. Send aggregated response
+                    aggregated_response_sender
+                        .send(Ok(current_aggregated_response.unwrap()))
+                        .map_err(|_| BlsAggregationServiceError::ChannelError)?;
+                    return Ok(());
+                },
+                signed_task_digest = signatures_rx.recv() =>{
+                    // New signature, aggregate it. If threshold is met, start window
+                    let Some(digest) = signed_task_digest else {
+                        return Err(BlsAggregationServiceError::SignaturesChannelClosed);
+                    };
+                    // check if the operator has already signed for this digest
+                    if aggregated_operators
+                        .get(&digest.task_response_digest)
+                        .map(|operators| {
+                            operators
+                                .signers_operator_ids_set
+                                .contains_key(&digest.operator_id)
+                        })
+                        .unwrap_or(false)
+                    {
+                        digest
+                            .signature_verification_channel
+                            .send(Err(SignatureVerificationError::DuplicateSignature))
+                            .await
+                            .map_err(|_| BlsAggregationServiceError::ChannelError)?;
+                        continue;
+                    }
+
+                    let verification_result = BlsAggregatorService::<A>::verify_signature(
+                        task_index,
+                        &digest,
+                        &operator_state_avs,
+                    )
+                    .await;
+                    let verification_failed = verification_result.is_err();
+
+                    digest
+                        .signature_verification_channel
+                        .send(verification_result)
+                        .await
+                        .map_err(|_| BlsAggregationServiceError::ChannelError)?;
+
+                    if verification_failed {
+                        continue;
```

Comment:
can we add a debug! log here with appropriate params ?

Reply:
Done!